src/core/config/config.rs

```diff
@@ -386,6 +386,9 @@ pub struct Arg {
     pub modify: Option<Modify>,
     #[serde(default, skip_serializing_if = "is_default")]
     pub default_value: Option<Value>,
+    /// Flag to indicate if the type inside the list is required.
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub list_type_required: bool,
```

Comment:
drop these changes if they aren't working from this PR. Reduce reviewer overhead :)